imagely.bin.js

```diff
@@ -39,10 +39,16 @@ if (options.log) {
 else if (options.batch) {
 	callback = function() {
 		if (this.jsonIndex < this.json.length) {
-			var name = this.destination.split('.').slice(0, -1).pop();
-			var uniqueName = this.destination.replace(name, name + this.jsonIndex);
-			var html = this.setWindowData(this.originalHtmlString, JSON.stringify(this.json[this.jsonIndex]));
+			var json = this.json[this.jsonIndex];
+			var uniqueName;
 
+			if (json.length > 0) {
+				var name = this.destination.split('/').slice(-1).pop();
```

Comment:
For filepath parsing, use Node's built-in `path` methods:
```js
var name = path.basename(this.destination);
```
You'll need to `var path = require('path');` above but don't need to `yarn add` anything—`path` is a native Node package.